yaas/repo.py

```diff
@@ -1,4 +1,99 @@
-# coding: utf-8
+from __future__ import absolute_import
+from __future__ import print_function
+
+from . import utils
 
 class Repo:
-    pass
+    """ Interact with Ambari stack repository configurations. """
+
+    def __init__(self, config):
+        self.config = config
+
+    def ls(self):
+        """ List all configured repositories. """
+        response = self.config.request(
+            'get',
+            '/api/v1/stacks',
+            params={'fields': 'versions/repository_versions/operating_systems/repositories/*'})
+        res = response.json()
+
+        repos = []
+        for stack in res['items']:
+            for version in stack['versions']:
+                for repo_version in version['repository_versions']:
+                    version = {
+                        'stack_name': repo_version['RepositoryVersions']['stack_name'],
+                        'stack_version': repo_version['RepositoryVersions']['stack_version'],
+                        'repo_id': repo_version['RepositoryVersions']['id'],
+                        'operating_systems': {}
+                        }
+                    for os in repo_version['operating_systems']:
+                        version['operating_systems'][os["OperatingSystems"]["os_type"]] = []
+                        for repo in os['repositories']:
+                            version['operating_systems'][os["OperatingSystems"]["os_type"]].append(repo['Repositories']['base_url'])
+                    repos.append(version)
+        return repos
+
+    def show(self, stack_name, stack_version, repo_id):
+        """ List all configured repositories. """
+        response = self.config.request(
+            'get',
+            '/api/v1/stacks/{0}/versions/{1}/repository_versions/{2}?fields=operating_systems/repositories/*'.format(
+                stack_name, stack_version, repo_id))
+        res = response.json()
+        utils.remove_hrefs(res)
```

Comment:
I can't help but think that it would only be a matter of time before someone wants access to the raw response... no action item here, just a note.